### PR TITLE
Pin dependencies

### PR DESCRIPTION
**Pull request by Adamantios**
## Proposed changes

Pins dependencies in `pipfile` and `tox` and removes some unnecessary dependencies from `tox`.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

n/a

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -426,14 +417,6 @@ usedevelop = True
 deps =
     grpcio==1.43.0
     py-eth-sig-utils==0.4.0
-    numpy==1.21.6
```
Comment (DavidMinarsch) [inline tox.ini:429]: nice!

File: tox.ini [hunk 2]
```diff
@@ -410,14 +409,6 @@ usedevelop = True
...
```
Comment (DavidMinarsch) [inline tox.ini:413]: Have you run `tox -e abci-docstrings` locally? It doesn't run in CI so cannot know from CI if this is still working.

Comment (Adamantios) [reply tox.ini:413]: Yes, still working. Not sure if there is something cached though :thinking:

**Review by dvilelaf — COMMENTED** [review]
LGTM, let's wait for tests.

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**